src/pip/_internal/locations.py

```diff
@@ -75,16 +75,16 @@ def get_src_prefix():
 except AttributeError:
     user_site = site.USER_SITE
 
+bin_user = sysconfig.get_path('scripts')
```

Comment:
This needs to use the `{os.name}_user` scheme instead of the default one. The default scheme points to `sys.prefix`, not the per-user location.

I would do something like

```python
def get_bin_user():
    scheme = "{}_user".format(os.name)
    if scheme not in sysconfig.get_scheme_names():
        scheme = "posix_user"  # Default to POSIX for unknown platforms.
    path = sysconfig.get_path("scripts", scheme=scheme)
    assert path is not None
    return path
```

This is also why the test suite is currently failing. The failing test is correctly detecting you trying to uninstall the script in system-bin instead of user-bin.